### PR TITLE
Initial build manifest for the Electron base app

### PR DESCRIPTION
**Pull request by cosimoc**

### CODE REVIEW

File: io.atom.electron.BaseApp.json [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    "build-options" : {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
+    "modules": [
+        {
+            "name": "udev",
```
Comment (TingPing) [inline io.atom.electron.BaseApp.json]: Can use the udev from `shared-modules`.

**Comment by TingPing** [comment]
Might want something like this:

```json
    {
      "name": "gvfs-trash",
      "buildsystem": "simple",
      "build-commands": [ "install -m755 gvfs-trash /app/bin/gvfs-trash" ],
      "sources": [{
        "type": "script",
        "dest-filename": "gvfs-trash",
        "commands": ["exec rm -r \"$@\""]
      }]
    }
```

As electron calls that but normal `gio trash` fails in the mounted fs.

**Comment by cosimoc** [comment]
@TingPing Fixed your comments, thanks.

**Comment by alexlarsson** [comment]
flathub now uses --install-deps-from=flathub when it builds, so it should automatically pull in the base app as needed, which will be great for electron based apps i believe. We should get this in so we can test a depdendency.

**Comment by Barthalion** [comment]
Test build here: https://flathub.org/builds/#/builders/1/builds/217

**Comment by TingPing** [comment]
Also next chromium bump in Electron we get Gtk3 which drops most of these deps, what will be the most elegant way to handle that other than just leaving unused libs here?

**Comment by TingPing** [comment]
And my `gvfs` comment still applies, you can't trash files on the host with gvfs so you have to delete them currently.

**Comment by Barthalion** [comment]
It's going to fail because of missing appdata. I guess it should be optional (and controlled with `flathub.json`).

**Comment by cosimoc** [comment]
@TingPing I forgot to comment about the trash -- what would require an Electron app to use the gvfs trash backend to trash files?

@Barthalion Is that option already available, or would we need to implement it on the Flathub side first?

**Comment by bil-elmoussaoui** [comment]
@TingPing I don't think that electron is going to ship chromium with gtk3 anytime soon. they are waiting for Chrome 62 I guess which is going to be released in September.  Once it's done, the apps must upgrade their electron dependency to the latest one (which might not happen for all the apps).
So I think, the best way to handle this is to have two versions of the electron base app. One that uses old electron with gtk2 stuff and the other one using electron + gtk3. And maybe in the future a newer version which will use gtk3 + dconf:+1:

**Comment by TingPing** [comment]
> what would require an Electron app to use the gvfs trash backend to trash files?

The `shell.moveItemToTrash()` method calls it, this is used by Atom when you delete files for example. Not sure if that is what you were asking.

**Comment by Barthalion** [comment]
@cosimoc: It is not AFAIK. I will try to look at it tomorrow and poke @ramcq if I fail.

**Comment by Barthalion** [comment]
Thanks to Alex's help, it's possible to skip appstream check now. Test build succeeded: https://flathub.org/builds/#/builders/1/builds/219

**Comment by alexlarsson** [comment]
Merged this, its building here: https://flathub.org/builds/#/builders/1/builds/220

**Comment by alexlarsson** [comment]
TingPing: I added an issue for the gvfs-trash thing, I just wanted to get this built so we can test the dependencies work.

**Pull request closed without merging**